### ISSUE TITLE
I was not able to train this model, loss was always 0.69 and ACC was always 0.5

### ISSUE THREAD

**Issue by Mr-watermelon** [brijeshiitg/XuNet-Structural-Design-of-Convolutional-Neural-Networksfor-Steganalysis#1]
Hello, author, 'prep = F.c onv2D (x, KV, padding=2)' is this sentence high-pass filtering mentioned in the article?Why can't I write my own training code to train the network?Can you upload a training code?

**Comment by Floating-dust** [comment]
I met the same problem. Have you solved it？

**Comment by Mr-watermelon** [comment]
> I met the same problem. Have you solved it？

No, have you solved it?

**Comment by Floating-dust** [comment]
Me too. @Mr-watermelon

**Comment by Mr-watermelon** [comment]
> Me too. @Mr-watermelon

I set the embedding rate to 0.8BPP, but the network still cannot converge. I don't know whether there is a problem with the network.

**Comment by Mr-watermelon** [comment]
> Me too. @Mr-watermelon

The author's model high-pass filtering does not know why /12, if this division is removed, the model can converge, but my results on the test set can only reach about 74

**Comment by brijeshiitg** [comment]
Hi guys,
Sorry for replying late. Use data augmentation: rotation, horizontal flip, and vertical flip while loading your images then your training should converge.

**Comment by Mr-watermelon** [comment]
> Hi guys,
> Sorry for replying late. Use data augmentation: rotation, horizontal flip, and vertical flip while loading your images then your training should converge.
Hello, brother, I have used data enhancement, but it still fails to converge well. For the test set with an embedding rate of 0.4 BPP, the accuracy can only reach about 74.However, the data set I used with the embedding rate of 0.8BPP can converge well. Does this indicate that the model itself has certain limitations?

**Comment by brijeshiitg** [comment]
> > Hi guys,
> > Sorry for replying late. Use data augmentation: rotation, horizontal flip, and vertical flip while loading your images then your training should converge.
> > Hello, brother, I have used data enhancement, but it still fails to converge well. For the test set with an embedding rate of 0.4 BPP, the accuracy can only reach about 74.However, the data set I used with the embedding rate of 0.8BPP can converge well. Does this indicate that the model itself has certain limitations?

Yes. maybe the model has some different hyperparameter settings.

**Comment by Mr-watermelon** [comment]
> > > Hi guys,
> > > Sorry for replying late. Use data augmentation: rotation, horizontal flip, and vertical flip while loading your images then your training should converge.
> > > Hello, brother, I have used data enhancement, but it still fails to converge well. For the test set with an embedding rate of 0.4 BPP, the accuracy can only reach about 74.However, the data set I used with the embedding rate of 0.8BPP can converge well. Does this indicate that the model itself has certain limitations?
> 
> Yes. maybe the model has some different hyperparameter settings.

I get it. Thank you, bro!

**Comment by Abhijithdidin** [comment]
Hello guys, can anyone share the training code

**Comment by brijeshiitg** [comment]
Training code is now available.
